python/utils/terra_utils/terra_workflow_configs.py

```diff
@@ -1,218 +1,223 @@
+        self.is_anvil = is_anvil
+        self.billing_project = billing_project
+
+        # Check if the workflow name is in the YAML file
+        available_workflows = GetWorkflowNames().get_workflow_names()
+        if workflow_name not in available_workflows:
+            raise ValueError(f"Workflow name {workflow_name} not found in {YAML_FILE_FULL_PATH}: {available_workflows}")
+
+        # Load the YAML file
+        with open(YAML_FILE_FULL_PATH, 'r') as file:
+            yaml_data = yaml.safe_load(file)
+        # Extract specific workflow information from yaml_data
+        self.yaml_info = next(workflow for workflow in yaml_data['workflows'] if workflow['name'] == self.workflow_name)
+        self.workflow_info = self._create_workflow_info_dict()
+        # Set the initial workflow configuration
+        self.workflow_config = self._create_up_workflow_config()
+
+    def _create_wdl_absolute_path(self, relative_path: str) -> str:
+        """
+        Create an absolute path from a relative path. Will join the relative path with the script's directory.
+
+        Args:
+            relative_path (str): The relative path to convert.
+
+        Returns:
+            str: The absolute path.
+        """
+        return os.path.join(WDL_ROOT_DIR_FULL_PATH, relative_path.lstrip('/'))
+
+    @staticmethod
+    def _get_wdl_workflow_name(wdl_file_path: str) -> str:
+        """
+        Get the WDL workflow name from the WDL file.
+
+        Args:
+            wdl_file_path (str): The path to the WDL file.
+
+        Returns:
+            str: The name of the WDL workflow.
+
+        Raises:
+            ValueError: If the workflow name is not found in the WDL file.
+        """
+        with open(wdl_file_path, 'r') as file:
+            for line in file:
+                # Search for the workflow name in the WDL file
+                match = re.search(r'^workflow\s+(\w+)\s', line)
+                if match:
+                    return match.group(1)
+        raise ValueError(f"Workflow name not found in {wdl_file_path}")
+
+    def _create_input_defaults(self) -> dict:
+        """
+        Set the default input values for the workflow configuration.
+        """
+        return {
+            f"{self.workflow_info['wdl_workflow_name']}.docker": f"\"{DOCKER_IMAGE}\"",
+            f"{self.workflow_info['wdl_workflow_name']}.max_retries": "5",
+            f"{self.workflow_info['wdl_workflow_name']}.max_backoff_time": "300",
+            f"{self.workflow_info['wdl_workflow_name']}.update_strategy": "\"REPLACE\"",
+            f"{self.workflow_info['wdl_workflow_name']}.bulk_mode": "false",
+            f"{self.workflow_info['wdl_workflow_name']}.workers": "10",
+            f"{self.workflow_info['wdl_workflow_name']}.batch_size": "500",
+            f"{self.workflow_info['wdl_workflow_name']}.batch_size_to_list_files": "20000",
+            f"{self.workflow_info['wdl_workflow_name']}.file_ingest_batch_size": "500",
         }
```

Comment:
Should we define all of these default values in an __init__ file somewhere and import them here, and also import them in all the python scripts? They're re-defined in each python script, and I'm worried that we might accidentally set a default that's different amongst the python scripts (see [here](https://github.com/broadinstitute/ops-terra-utils/blob/b000635f19708b480e28025db8b6f0017a692bba/python/gcp_workspace_table_to_dataset_ingest.py#L26) and [here](https://github.com/broadinstitute/ops-terra-utils/blob/b000635f19708b480e28025db8b6f0017a692bba/python/get_and_delete_orphaned_files_from_dataset.py#L14) as examples)

Reply:
This can also maybe be a separate PR since this one is already pretty big, but just a thought to help us keep things consistent

Reply:
Good idea, adding now